jetty-http/src/main/java/org/eclipse/jetty/http/MultiPartParser.java

```diff
@@ -694,16 +723,17 @@ default void earlyEOF()
         {
         }
     }
-    
+
     /* ------------------------------------------------------------------------------- */
     @SuppressWarnings("serial")
-    private static class IllegalCharacterException extends IllegalArgumentException
+    private static class IllegalCharacterException extends BadMessageException
```

Comment:
This class is used in both `HttpParser` and here, no point in duplicating, make it public in its own file.

Reply:
The class is similar, but subtly different.  It has a different parent class and a different state that it reports